src/org/sosy_lab/java_smt/api/Model.java

```diff
@@ -52,11 +52,15 @@ public interface Model extends Evaluator, Iterable<ValueAssignment>, AutoCloseab
    */
   @Override
   default Iterator<ValueAssignment> iterator() {
-    return asList().iterator();
+    try {
+      return asList().iterator();
+    } catch (InterruptedException pE) {
+      throw new RuntimeException(pE);
```

Comment:
It would be nice to hide the runtime eexception here and just forward the InterruptedException, e.g., without changing the stacktrace and interrupt the exception trace.

Reply:
ignore this